### PR TITLE
HS-1219: add pagination to alerts/listAlerts

### PR DESCRIPTION
**Pull request by BenjaminJ**
## Description
add pagination to alerts/listAlerts
## Jira link(s)
- https://issues.opennms.org/browse/HS-1219

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts

### CODE REVIEW

File: alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java [hunk 1]
```diff
@@ -52,13 +56,32 @@ public class AlertGrpcService extends AlertServiceGrpc.AlertServiceImplBase {
 
     @Override
     public void listAlerts(ListAlertsRequest request, StreamObserver<ListAlertsResponse> responseObserver) {
-        List<Alert> alerts = alertRepository.findAll().stream()
+        // Extract the page size and next page token values from the request
+        int pageSize = request.getPageSize() != 0 ? request.getPageSize() : 10; // default page size is 10
```
Comment (cgorantla) [inline alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java:60]: Use constants for page size and default page number

File: alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java [hunk 2]
```diff
@@ -52,13 +56,32 @@ public class AlertGrpcService extends AlertServiceGrpc.AlertServiceImplBase {
...
+        String nextPageToken = !request.getNextPageToken().isEmpty() ? request.getNextPageToken() : "0"; // default page is 0
+
+        // Create a PageRequest object based on the page size and next page token
+        PageRequest pageRequest = PageRequest.of(0, pageSize);
+        if (!nextPageToken.isEmpty()) {
```
Comment (cgorantla) [inline alert/src/main/java/org/opennms/horizon/alertservice/grpc/AlertGrpcService.java:65]: Isn't this always true ?

File: alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java [hunk 3]
```diff
@@ -206,9 +206,10 @@ public void sendGETRequestToApplicationAtPathUntilJSONResponseMatchesTheFollowin
     }
 
     @Then("Verify alert topic has {int} messages with tenant {string}")
-    public void verifyTopicContainsTenant(int expectedMessages, String tenant) {
+    public void verifyTopicContainsTenant(int expectedMessages, String tenant) throws InterruptedException {
         int foundMessages = 0;
 
+        Thread.sleep(1000);
```
[the remark this answers is not in the document]
Comment (BenjaminJ) [reply alert/src/test/java/org/opennms/horizon/alertservice/stepdefs/AlertTestSteps.java:212]: oops didn't want to commit this

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**